src/ATen/native/xpu/sycl/SortingKernels.h

```diff
@@ -397,10 +407,32 @@ void segmented_radix_sort_pairs_kernel(
       std::swap(keys_in_, keys_out_);
       std::swap(values_in_, values_out_);
     }
+    result_in_temp = !result_in_temp;
```

Comment:
There is no other logic except for `result_in_temp = false` before it? Using `!result_in_temp` makes confusion.
```
result_in_temp = true;
```

Reply:
This is the basic operation of the AB Buffer, where the inverse is taken at each step.